File: BoboTu/BoboTu.Data/Models/Venue.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace BoboTu.Data.Models
{
    public class Venue
    {
        public Venue()
        {
            Ratings = new HashSet<Rating>();
            Opinions = new HashSet<Opinion>();
            VenueFacilities = new HashSet<VenueFacility>();

        }

        public int Id { get; set; }
        public string Name { get; set; }

        public double Lattitude { get; set; }
        public double Longitude { get; set; }
        public string City { get; set; }
        public string Street { get; set; }
        public int HouseNumber { get; set; }
        public string ZipCode { get; set; }
        public VenueType VenueType { get; set; }
        public  ICollection<Rating> Ratings { get; set; }
        public  ICollection<Opinion> Opinions { get; set; }
        public  ICollection<VenueFacility> VenueFacilities { get; set; }

        public string Description { get; set; }
        public double AverageRating { get; set; }

        public string WebPage { get; set; }
        public string EmailAddress { get; set; }
    }
}

File: BoboTu.Cllent/src/app/Models/Venue.ts
import { Rating } from "./Rating"
import { Opinion } from "./Opinion"
import { VenueType, FacilityType } from "./Enums"

export class Venue{
     id:number 
      name :string       
      city :string
      street :string
      zipCode :string
     venueType: VenueType 
     ratings: Array<Rating>  
     opinions:   Array<Opinion>
      description: string 
     averageRating:number 
      webPage: string 
      houseNumber:number
      emailAddress: string 
      facilities:Array<any>
      countOfValueRatings:Array<CountOfValueRating>
       lattitude:number
       longitude:number;
       ratingsCount:number;
     
}

export class CountOfValueRating{
     ratingValue:number
     count:number;
}

